src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -181,34 +182,34 @@ public static ClassifiedSpan[] ConvertMultiLineToSingleLineSpans(SourceText text
                 var numLinesInSpan = endLine - startLine + 1;
                 Contract.ThrowIfTrue(numLinesInSpan < 1);
 
-                var updatedSpanIndex = spanIndex;
-
                 for (var currentLine = 0; currentLine < numLinesInSpan; currentLine++)
                 {
-                    TextSpan? textSpan;
+                    TextSpan textSpan;
+                    var line = text.Lines[startLine + currentLine];
 
                     // Case 1: First line of span
                     if (currentLine == 0)
                     {
-                        var absoluteStartOffset = text.Lines[startLine].Start + startOffset;
-                        var spanLength = text.Lines[startLine].End - absoluteStartOffset;
-                        textSpan = new TextSpan(absoluteStartOffset, spanLength);
+                        var absoluteStart = line.Start + startOffset;
+
+                        // This start could be past the regular end of the line if it's within the newline character if we have a CRLF newline. In that case, just skip emitting a span for the LF.
```

Comment:
a mention of what scenario the start is passed the regular end of line would be useful here (i.e. when the regex parser outputs this)

Reply:
Added comment, but at the same time I'm considering this a fix for the scenario generally; it wouldn't surprise me if there's more ways to this we haven't discovered yet.